app/controllers/scan_analytics_controller.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+class ScanAnalyticsController < ApplicationController
+  helper ProjectsHelper
+
+  before_action :set_project_or_fail
+  before_action :analytics_data, only: :index
+
+  def index
+    return if @analytics.blank?
+
+    params[:code_set_id] = params[:code_set_id] || @analytics.first.code_set_id
+    @scan_data = @analytics.where(code_set_id: params[:code_set_id]).first
+  end
+
+  def charts
+    return unless @project.coverity_project_id
+
+    scan_project_id = params[:scan_id].presence || @project.coverity_project_id
+    api_response(scan_project_id)
+  end
+
+  private
+
+  def analytics_data
+    @analytics = @project.best_analysis&.scan_analytics&.order(id: :asc) if @project.best_analysis
```

Comment:
`scan_analytics` is an array. When there are no results, it will be an empty array, which will not raise an exception for `order`. Hence we do not need `&.` after `scan_analytics`. One never needs to use `&.` with an array.

Reply:
Why are we doing `if @project.best_analysis` while also doing `@project.best_analysis` with an `&.`? You need either one of those, not both.

Reply:
I got an rubocop error after removing & after scan_analytics.
Below is the error I got it
https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Lint/SafeNavigationChain